pkg/retrieval/retrieval.go

```diff
@@ -275,10 +304,13 @@ func (s *Service) retrieveChunk(ctx context.Context, peer swarm.Address, done ch
 	// Reserve to see whether we can request the chunk
 	creditAction, err := s.accounting.PrepareCredit(creditCtx, peer, chunkPrice, isOrigin)
 	if err != nil {
+		skip.Add(addr, peer, overDraftRefresh)
 		return
 	}
 	defer creditAction.Cleanup()
 
+	skip.Add(addr, peer, maxDuration)
```

Comment:
We add the overdraft peers separately. If there is an error with this request, we will add the peer to the errSkip list in retrieveChunk. If the request was successful we will have the chunk. So I am not sure if there is any point in adding the peer to the local skiplist here. We might end up wasting some CPU to check a list with duplicate peers in the closestPeer function.

Reply:
if we add the peer to the skiplist at the end of the request, a preemtive attempt could try the same peer before the first request returns.